vault/namespace_store.go

```diff
@@ -874,61 +904,49 @@ func (ns *NamespaceStore) UnlockNamespace(ctx context.Context, unlockKey, path s
 		return fmt.Errorf("namespace %q is not locked", namespaceToUnlock.Path)
 	}
 
-	lockingNamespace := ns.GetLockingNamespace(namespaceToUnlock)
+	lockingNamespace := ns.getLockingNamespace(namespaceToUnlock)
 	if lockingNamespace.ID != namespaceToUnlock.ID {
 		return fmt.Errorf("cannot unlock %q with namespace %q being locked", namespaceToUnlock.Path, lockingNamespace.Path)
 	}
 
-	ns.lock.Lock()
-	return ns.unlockNamespaceLocked(ctx, namespaceToUnlock, unlockKey)
-}
-
-// unlockNamespaceLocked reads namespace lock from the storage, compares it to the
-// provided unlock key, with match deletes the storage entry, modifies namespace
-// tree structure and namespace maps changing the 'Locked' field to false.
-func (ns *NamespaceStore) unlockNamespaceLocked(ctx context.Context, target *namespace.Namespace, unlockKey string) error {
-	unlock := true
-	defer func() {
-		if unlock {
-			ns.lock.Unlock()
-		}
-	}()
-
-	item, ok := ns.namespacesByUUID[target.UUID]
-	if !ok {
-		return fmt.Errorf("unable to unlock unknown namespace %v", target.ID)
-	}
-
 	// verify lock or skip when provided unlock key is empty
 	// (meaning namespace is unlocked using root capability)
-	if unlockKey != "" && subtle.ConstantTimeCompare([]byte(item.UnlockKey), []byte(unlockKey)) != 1 {
+	if unlockKey != "" &&
```

Comment:
```suggestion
	if len(unlockKey) > 0 &&
```

Wonder if this is safer than a nil check from a timing attack perspective.

Reply:
Go is smart enough to compile these to the same assembly.